lib/internal/test_runner/test.js

```diff
@@ -404,6 +404,7 @@ class Test extends AsyncResource {
         kCancelledByParent
       )
     );
+    this.error.failureType = kCancelledByParent;
```

Comment:
I think we need to do this conditionally if the failure type is not already set. Looking at the changes to the `.out` changes in this PR, it appears that `testTimeoutFailure`s are being overwritten.

Reply:
not sure I follow how you suggest handling this, we want to know if a test is canceled and it seems fine to have a single `failureType` for all canceled tests, doesn't it?

Reply:
A test can be "cancelled" for a few reasons (based on calling the test's `cancel()` method, which sets the test's `cancelled` flag): test timeout, the parent test finishing first, abort signal, etc. In the summary at the end of the test run, we want all of those to be grouped as cancelled. But when showing information for individual tests, we want to give the more granular information.